src/main/java/fi/helsinki/opintoni/localization/TeachingLanguages.java

```diff
@@ -20,91 +20,114 @@
+    PT(Code.PT, ImmutableMap.of(
         Language.FI.getCode(), "portugali",
         Language.SV.getCode(), "portugisiska",
         Language.EN.getCode(), "Portuguese")),
-    FR("fr", ImmutableMap.of(
+    FR(Code.FR, ImmutableMap.of(
         Language.FI.getCode(), "ranska",
         Language.SV.getCode(), "franska",
         Language.EN.getCode(), "French")),
-    DE("de", ImmutableMap.of(
+    DE(Code.DE, ImmutableMap.of(
         Language.FI.getCode(), "saksa",
         Language.SV.getCode(), "tyska",
         Language.EN.getCode(), "German")),
-    DA("da", ImmutableMap.of(
+    DA(Code.DA, ImmutableMap.of(
         Language.FI.getCode(), "tanska",
         Language.SV.getCode(), "danska",
         Language.EN.getCode(), "Danish")),
-    RU("ru", ImmutableMap.of(
+    RU(Code.RU, ImmutableMap.of(
         Language.FI.getCode(), "venäjä",
         Language.SV.getCode(), "ryska",
         Language.EN.getCode(), "Russian")),
-    ET("et", ImmutableMap.of(
+    ET(Code.ET, ImmutableMap.of(
         Language.FI.getCode(), "viro",
         Language.SV.getCode(), "estniska",
         Language.EN.getCode(), "Estonian")),
-    FI_1ST_SC("fi_1st_sc", ImmutableMap.of(
+    FI_1ST_SC(Code.FI_1ST_SC, ImmutableMap.of(
         Language.FI.getCode(), "äidinkieli (suomi), puheviestintä- ja vuorovaikutustaidot",
         Language.SV.getCode(), "modersmål (finska), retorik",
         Language.EN.getCode(), "Finnish as a native language, speech communication")),
-    FI_1ST_AW("fi_1st_aw", ImmutableMap.of(
+    FI_1ST_AW(Code.FI_1ST_AW, ImmutableMap.of(
         Language.FI.getCode(), "äidinkieli (suomi), akateemiset tekstitaidot",
         Language.SV.getCode(), "modersmål (finska), akademiskt skrivande",
         Language.EN.getCode(), "Finnish as a native language, academic writing")),
-    FI_2ND("fi_2nd", ImmutableMap.of(
+    FI_2ND(Code.FI_2ND, ImmutableMap.of(
         Language.FI.getCode(), "suomi toisena kotimaisena kielenä (finska)",
         Language.SV.getCode(), "finska som andra inhemska språk",
         Language.EN.getCode(), "Finnish for native speakers of Swedish")),
-    SV_1ST("sv_1st", ImmutableMap.of(
+    SV_1ST(Code.SV_1ST, ImmutableMap.of(
         Language.FI.getCode(), "äidinkieli (ruotsi)",
         Language.SV.getCode(), "modersmål (svenska)",
         Language.EN.getCode(), "Swedish as a native language")),
-    SV_2ND("sv_2nd", ImmutableMap.of(
+    SV_2ND(Code.SV_2ND, ImmutableMap.of(
         Language.FI.getCode(), "ruotsi toisena kotimaisena kielenä",
         Language.SV.getCode(), "svenska som andra inhemska språk",
         Language.EN.getCode(), "Swedish for native speakers of Finnish")),
-    SLU("slu", ImmutableMap.of(
+    SLU(Code.SLU, ImmutableMap.of(
         Language.FI.getCode(), "pedagoginen yliopistonlehtori",
         Language.SV.getCode(), "pedagogisk universitetslektor",
         Language.EN.getCode(), "senior lecturer in university pedagogy"));
 
-    private final String code;
+    public enum Code {
```

Comment:
Ei tarvita sisempää enumia koodin mallintamiseen. Koodi voi olla enumissa string, ja sallitut arvot luodaan enumin bodyssä normaalisti. Entiteetin kentän tyypiksi taas asetetaan itse enum, ja enumille kirjoitetaan JPA-converter joka muuntaa arvon kantaan kirjoitettaessa enumin sisälmäksi koodiksi ja kannasta lukiessa itse enumiksi. Ks. https://thoughts-on-java.org/jpa-21-type-converter-better-way-to/